### PR TITLE
minigun rework makes it work off powerpacks, also deletes old sg

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
replaces old sg with a new minigun that works off powerpacks to change the old minigun to work off new minigun powerpacks that reload the minigun
also gives it mag harness considering it'd use all 3 of you slots to make the minigun not entirely useless with a belt harness, as you now have to give up your back slot

## Why It's Good For The Game
makes the minigun more fun and interesting and deletes an ancient unnededed cm thing

## Changelog
:cl:
del: Removed old m56
balance: minigun now has magnetic harness, and works off powerpacks that reload it instead of ammo magazines
/:cl:

### CODE REVIEW

File: code/game/objects/items/storage/marine_boxes.dm [hunk 1]
```diff
@@ -37,9 +37,9 @@
 	new /obj/item/storage/belt/marine/t25(src)
 	new /obj/item/ammo_magazine/rifle/standard_smartrifle(src)
 
-/obj/item/smartgun_powerpack
+/obj/item/minigun_powerpack
 	name = "\improper M56 powerpack"
-	desc = "A heavy reinforced backpack with support equipment, power cells, and spare rounds for the M56 Smartgun System.\nClick the icon in the top left to reload your M56."
+	desc = "A heavy reinforced backpack with support equipment, power cells, and spare rounds for the M56 minigun System.\nClick the icon in the top left to reload your M56."
```
Comment (Pariah919) [inline code/game/objects/items/storage/marine_boxes.dm:42]:
```suggestion
	desc = "A heavy reinforced backpack with support equipment, power cells, and spare rounds for the T-100 Minigun.\nClick the icon in the top left to reload your M56."
```

File: code/game/objects/items/storage/marine_boxes.dm [hunk 2]
```diff
@@ -52,18 +52,18 @@
...
+/obj/item/minigun_powerpack/attack_self(mob/living/carbon/human/user, automatic = FALSE)
 	if(!istype(user) || user.incapacitated())
 		return FALSE
 
-	var/obj/item/weapon/gun/smartgun/mygun = user.get_active_held_item()
+	var/obj/item/weapon/gun/minigun/mygun = user.get_active_held_item()
 
 	if(!istype(mygun))
-		to_chat(user, "You must be holding an M56 Smartgun to begin the reload process.")
+		to_chat(user, "You must be holding an M56 minigun to begin the reload process.")
```
Comment (Pariah919) [inline code/game/objects/items/storage/marine_boxes.dm:66]:
```suggestion
		to_chat(user, "You must be holding an T-100 Minigun to begin the reload process.")
```

File: code/game/objects/items/storage/marine_boxes.dm [hunk 3]
```diff
@@ -82,9 +82,9 @@
 
 	reloading = TRUE
 	if(!automatic)
-		user.visible_message("[user.name] begins feeding an ammo belt into the M56 Smartgun.","You begin feeding a fresh ammo belt into the M56 Smartgun. Don't move or you'll be interrupted.")
+		user.visible_message("[user.name] begins feeding an ammo belt into the M56 minigun.","You begin feeding a fresh ammo belt into the M56 minigun. Don't move or you'll be interrupted.")
```
Comment (Pariah919) [inline code/game/objects/items/storage/marine_boxes.dm:85]:
```suggestion
		user.visible_message("[user.name] begins feeding an ammo belt into the M56 minigun.","You begin feeding a fresh ammo belt into the T-100 Minigun. Don't move or you'll be interrupted.")
```

File: code/game/objects/items/storage/marine_boxes.dm [hunk 4]
```diff
@@ -82,9 +82,9 @@
...
 	else
-		user.visible_message("[user.name]'s powerpack servos begin automatically feeding an ammo belt into the M56 Smartgun.","The powerpack servos begin automatically feeding a fresh ammo belt into the M56 Smartgun.")
+		user.visible_message("[user.name]'s powerpack servos begin automatically feeding an ammo belt into the M56 minigun.","The powerpack servos begin automatically feeding a fresh ammo belt into the M56 minigun.")
```
Comment (Pariah919) [inline code/game/objects/items/storage/marine_boxes.dm:87]:
```suggestion
		user.visible_message("[user.name]'s powerpack servos begin automatically feeding an ammo belt into the M56 minigun.","The powerpack servos begin automatically feeding a fresh ammo belt into the T-100 Minigun.")
```

File: code/game/objects/items/storage/marine_boxes.dm [hunk 5]
```diff
@@ -151,15 +151,15 @@
 	rounds_remaining -= rounds_to_reload
 
 	if(!automatic)
-		to_chat(user, "You finish loading [rounds_to_reload] shells into the M56 Smartgun. Ready to rumble!")
+		to_chat(user, "You finish loading [rounds_to_reload] shells into the M56 minigun. Ready to rumble!")
```
Comment (Pariah919) [inline code/game/objects/items/storage/marine_boxes.dm:154]:
```suggestion
		to_chat(user, "You finish loading [rounds_to_reload] shells into the T-100 Minigun. Ready to rumble!")
```

File: code/game/objects/items/storage/marine_boxes.dm [hunk 6]
```diff
@@ -151,15 +151,15 @@
...
 	else
-		to_chat(user, "The powerpack servos finish loading [rounds_to_reload] shells into the M56 Smartgun. Ready to rumble!")
-	playsound(user, 'sound/weapons/guns/interact/smartgun_unload.ogg', 25, 1)
+		to_chat(user, "The powerpack servos finish loading [rounds_to_reload] shells into the M56 minigun. Ready to rumble!")
```
Comment (Pariah919) [inline code/game/objects/items/storage/marine_boxes.dm:156]:
```suggestion
		to_chat(user, "The powerpack servos finish loading [rounds_to_reload] shells into the T-100 Minigun. Ready to rumble!")
```

File: code/game/objects/items/storage/marine_boxes.dm [hunk 7]
```diff
@@ -574,9 +574,8 @@
 	new /obj/item/clothing/head/helmet/marine/specialist(src)
 	new /obj/item/weapon/gun/minigun(src)
 	new /obj/item/belt_harness/marine(src)
-	new /obj/item/ammo_magazine/minigun(src)
-	new /obj/item/ammo_magazine/minigun(src)
-	new /obj/item/ammo_magazine/minigun(src)
+	new /obj/item/minigun_powerpack(src)
+
```
Comment (Pariah919) [inline code/game/objects/items/storage/marine_boxes.dm:578]:
```suggestion
```

**Review by Pariah919 — COMMENTED** [review]
minigun name and stuff

File: code/modules/projectiles/guns/specialist.dm [hunk 8]
```diff
@@ -341,84 +341,113 @@
...
+		windup_checked = WEAPON_WINDUP_NOT_CHECKED
+		return
+	windup_checked = WEAPON_WINDUP_CHECKED
+	SEND_SIGNAL(src, COMSIG_GUN_FIRE)
+
+/obj/item/weapon/gun/minigun/get_ammo_type()
+	if(!ammo)
+		return list("unknown", "unknown")
+	else
+		return list(ammo.hud_state, ammo.hud_state_empty)
```
Comment (TiviPlus) [inline code/modules/projectiles/guns/specialist.dm:419]:
```suggestion
	return list(ammo.hud_state, ammo.hud_state_empty)
```

File: code/modules/projectiles/guns/specialist.dm [hunk 9]
```diff
@@ -341,84 +341,113 @@
...
+
+/obj/item/weapon/gun/minigun/get_ammo_count()
+	if(!current_mag)
+		return in_chamber ? 1 : 0
+	else
+		return in_chamber ? (current_mag.current_rounds + 1) : current_mag.current_rounds
```
Comment (TiviPlus) [inline code/modules/projectiles/guns/specialist.dm:425]:
```suggestion
		return in_chamber ? 1 : 0
	return in_chamber ? (current_mag.current_rounds + 1) : current_mag.current_rounds
```

File: code/modules/projectiles/magazines/specialist.dm [hunk 10]
```diff
@@ -263,6 +263,25 @@
 	default_ammo = /datum/ammo/rocket/wp/quad
 	max_rounds = 4
 
+
+/obj/item/ammo_magazine/internal/launcher/rocket/m57a4/ds
+	default_ammo = /datum/ammo/rocket/wp/quad/ds
+
+//-------------------------------------------------------
```
Comment (TiviPlus) [inline code/modules/projectiles/magazines/specialist.dm]: delete this

File: code/modules/projectiles/magazines/specialist.dm [hunk 11]
```diff
@@ -263,6 +263,25 @@
...
+	name = "rotating ammo drum (7.62x51mm)"
+	desc = "A huge ammo drum for a huge gun."
+	caliber = CALIBER_762X51
+	icon_state = "minigun"
+	default_ammo = /datum/ammo/bullet/minigun
+	max_rounds = 300
+	reload_delay = 50 //Hard to reload.
+	w_class = WEIGHT_CLASS_NORMAL
+	gun_type = /obj/item/weapon/gun/minigun
+	icon_state_mini = "mag_minigun"
```
Comment (TiviPlus) [inline code/modules/projectiles/magazines/specialist.dm]: did you add this on accident

Comment (Pariah919) [reply code/modules/projectiles/magazines/specialist.dm]: yes

**Comment by SplinterGP** [comment]
Give it spinning up and its epic

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**